pkg/scheduler/actions/backfill/backfill.go

```diff
@@ -133,10 +134,35 @@ func backFill(ssn *framework.Session, job *api.JobInfo) {
 			}
 
 			if task.InitResreq.LessEqual(node.Idle) {
+
+				/* patch the annotation */
+				annotation := map[string]string{v1alpha1.BackfillAnnotationKey: "true"}
+				err := ssn.PatchAnnotation(task, annotation)
+				if err != nil {
+					glog.Errorf("Failed to patch backfill=true to task/pod <%s/%s>: %v, will not backfill on this node %s",
+						task.Name, task.Pod.Name, err, node.Name)
+					continue
```

Comment:
We have a tough question here. Suppose the job have 5 tasks. What should we do, if we successfully patch the first 3 tasks, but the patch fails on the 4th task?  

I think we need to make sure that if patch fails on any task, we will not proceed to bind any task. In this case, some tasks may have annotations, but no task will be bound. For tasks that have annotations but are not bound to any host, we can remove their annotation in the next scheduling round when we create TaskInfo from Pod.

Reply:
that brings up another issue. what if dispatch fails on the 4th task?

Reply:
According to the current implementation, if dispatch fails on the 4th task, the scheduler will error out without dispatching the 5th task. 

Whether or not this is the correct behavior is debatable. But, at least, our patch logic does not make things worse. We may end up having a few tasks which have annotations but are not bounded. But this is okay because we are able to fix it in the next scheduling round.